### ISSUE TITLE
Gesture button press and release events

### ISSUE THREAD

**Issue by knzivid** [pwr-Solaar/Solaar#1186]
**Information**

- Solaar version: 1.0.6
- Distribution: Arch Linux
- Kernel version: Linux 5.12.10-arch1-1 x86_64 GNU/Linux

**Is your feature request related to a problem? Please describe.**

I use an MX Master 3. I would like to bind press and release events of the mouse gesture button `00C3` to separate commands. Is it possible? If so, how can I add custom tests called `mouse_gesture_pressed` and `mouse_gesture_down`? 

**Describe alternatives you've considered**

I tried using `Key` condition but that only responds on `press` and not `release`.

**Comment by pfps** [comment]
This is possible and not too hard, but might not be implemented for some time.

**Comment by nishanthkarthik** [comment]
As an experiment, I diverted the left button and right button events on my M330 mouse. In this exercise, I want to execute custom commands to the `pressed` and `released` notifications. 

```py
# left button pressed
feature = {NamedInt} REPROG CONTROLS V4
notification = {_HIDPP_Notification: 5} _HIDPP_Notification(report_id=17, devnumber=1, sub_id=13, address=0, data=b'\x00P\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')

# left button released
feature = {NamedInt} REPROG CONTROLS V4
notification = {_HIDPP_Notification: 5} _HIDPP_Notification(report_id=17, devnumber=1, sub_id=13, address=0, data=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')

# right button pressed
feature = {NamedInt} REPROG CONTROLS V4
notification = {_HIDPP_Notification: 5} _HIDPP_Notification(report_id=17, devnumber=1, sub_id=13, address=0, data=b'\x00Q\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')

# right button released
feature = {NamedInt} REPROG CONTROLS V4
notification = {_HIDPP_Notification: 5} _HIDPP_Notification(report_id=17, devnumber=1, sub_id=13, address=0, data=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')
```

The `notification.data` field lets me distinguish left button and right button in the `pressed` notification. The second byte is `P` vs `Q`. But, the `released` event sends the same `notification.data` for both left and right button - `\x00`. I took a stab at implementing a custom test in `TESTS` section on `diversion.py` but I got stuck trying to distinguish the `released` events. Any pointers would be appreciated :)

**Comment by pfps** [comment]
The implementation has to keep track of which buttons were pressed to determine which button was released.  The implementation already does this.  The code is in the process_notification function in diversion.py

**Comment by nishanthkarthik** [comment]
Thank you. I have a working prototype now. May I make a draft PR?

**Comment by pfps** [comment]
Sure.  Please include documentation and label the commit similar to other commits.

**Comment by pfps** [comment]
Fixed by #1189
